src/roles/piper.py

```diff
@@ -196,10 +219,19 @@ def on_acted(evt, var, target, spy):
 def on_reset(evt, var):
     CHARMED.clear()
     TOBECHARMED.clear()
+    PASSED.clear()
```

Comment:
Need to handle `PASSED` (and `TOBECHARMED` too, apparently) in the `exchange_roles` listener. Make sure to test as well.